pubsub/rabbitmq/rabbitmq.go

```diff
@@ -101,14 +101,22 @@ type rabbitMQConnectionBroker interface {
 }
 
 type ErrorCollection struct {
-	errors []error
-	mux    sync.Mutex
+	errors    []error
+	mux       sync.Mutex
+	ErrNotify chan struct{}
 }
 
-func (c *ErrorCollection) Append(e error, stopCh chan struct{}) {
+func NewErrorCollection() ErrorCollection {
+	return ErrorCollection{
+		errors:    []error{},
+		ErrNotify: make(chan struct{}),
+	}
+}
```

Comment:
You should return a `*ErrorCollection` here since your struct has a `sync.Mutex` field.

```
func NewErrorCollection() *ErrorCollection {
	return &ErrorCollection{
		errors:    []error{},
		ErrNotify: make(chan struct{}),
	}
}
```